### PR TITLE
Feature/image backend

### PR DESCRIPTION
**Pull request by BoostedBoat**
Added image backend + communication between the app and firestore/cloud storage.

### CODE REVIEW

**Comment by rkuenzi-epfl** [comment]
Minimal SDK bumped to 24!
Getting ExistingPicture require you to load them, which will return a CompletableFuture so you know when they are available.
(Use thenApply, or just get() if you don't want to do anything else in the mean time)
Picture are also cached so that you don't get them from cloud/firestore storage every time you want the same picture.
This cache might consume too much memory because of current image storage in the form of bitmaps so for now we should probably make sure to remove the images from cache when they are no longer necessary.

File: app/src/main/java/com/github/wnder/ExistingPicture.java [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    //Storage
+    private Storage storage;
+
+    //Image unique ID
+    private String uniqueId;
+    private Bitmap bmp;
+
+    //Image location
+    private long longitude;
+    private long latitude;
```
Comment (JeremyDezalos) [inline app/src/main/java/com/github/wnder/ExistingPicture.java:31]: You store 2 longs here and you wrote a getter that returns a LatLng. Why not store a LatLng right now ?

File: app/src/main/java/com/github/wnder/NewPicture.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    //User id
+    private String user;
+
+    //Image unique ID
+    private String uniqueId;
+    private Uri uri;
+
+    //Image location
+    private long longitude;
+    private long latitude;
```
Comment (JeremyDezalos) [inline app/src/main/java/com/github/wnder/NewPicture.java:30]: same

**Review by JeremyDezalos — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/wnder/Picture.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+     * Returns the unique ID of the picture
+     * @return The unique ID of the picture
+     */
+    public String getUniqueId();
+
+    /**
+     * Returns the longitude and latitude of the location of the picture
+     * @return the longitude and latitude of the location of the picture
+     */
+    public LatLng getLatLng();
```
Comment (stblr) [inline app/src/main/java/com/github/wnder/Picture.java:20]: Location should be used instead of LatLng

Comment (stblr) [inline app/src/main/java/com/github/wnder/Picture.java:20]: Location should be used instead of LatLng

**Review by stblr — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/wnder/Picture.java (hunk 3 above)
Comment (rkuenzi-epfl) [reply app/src/main/java/com/github/wnder/Picture.java]: Done

File: app/src/main/java/com/github/wnder/ExistingPicture.java [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    }
+
+    private void setBitmap(Bitmap bmp){
+        this.bmp = bmp;
+    }
+
+    private void setLocation(long longitude, long latitude){
+        this.longitude = longitude;
+        this.latitude = latitude;
+    }
```
[the remark this answers is not in the document]
Comment (rkuenzi-epfl) [reply app/src/main/java/com/github/wnder/ExistingPicture.java]: Done

**Review by stblr — APPROVED** [review]

**Review by JeremyDezalos — APPROVED** [review]

**Pull request merged**